R/metadata.R

```diff
@@ -114,57 +115,91 @@ metadata <- function(surveyID,
   # RESHAPE DATA ----
 
   # Metadata
-  metadata <- data.frame(
-    "surveyID" = resp_filt$id,
-    "name" = resp_filt$name,
-    "ownerId" = resp_filt$ownerId,
-    "organizationId" = resp_filt$organizationId,
-    "isActive" = resp_filt$isActive,
-    "creationDate" = resp_filt$creationDate,
-    "lastModifiedDate" = resp_filt$lastModifiedDate,
-    "expiration_startdate" = ifelse(is.null(resp_filt$expiration$startDate),
-      NA,
-      resp_filt$expiration$startDate
-    ),
-    "expiration_endDate" = ifelse(is.null(resp_filt$expiration$endDate),
-      NA,
-      resp_filt$expiration$endDate
+  if (survey_definition) {
+    metadata <- data.frame(
+      "surveyID" = resp_filt$SurveyID,
+      "name" = resp_filt$SurveyName,
+      "ownerId" = resp_filt$OwnerID,
+      "organizationId" = resp_filt$CreatorID,
+      "isActive" = resp_filt$SurveyStatus
+    )
```

Comment:
There is a fair amount of repetition here and in this section. Pull out anything that is the same for both endpoints, out of the if statement, to make it easier to maintain this code.

Reply:
The lists for the two endpoints have slightly different names, for example the list for the "surveys" endpoint stores survey ID in `surveyID` whereas for the "surveys_definition" it is stored in `SurveyID`. So this is technically not a repetition?

Reply:
Ah, I see what you're saying. When you get around to editing this PR, see if you can do any refactoring at all to improve this section, but if not we'll see what we can later.